### DISCUSSION TITLE
Failed to decode output: Lesson 4

### DISCUSSION THREAD

**Discussion by swayamarora5745** [Cyfrin/foundry-full-course-cu#1171] [Q&A] [answered]
Getting below error:
{
	"error": "Failed to decode output: Error: hex data is odd-length (argument=\"value\", value=\"0x0\", code=INVALID_ARGUMENT, version=bytes/5.7.0)"
}

in getPrice() method in Lesson 4 in the below lines:
(, int256 answer, , , ) = priceFeed.latestRoundData();
return answer;

any suggestions?

**Comment by 0xusmanf** [comment]
Can you please provide your solidity code here??

**Reply by swayamarora5745** [reply]
thanks for responding. Just after posting this question, I figured out what was wrong. I was deploying my code on the "Remix" environment in place of "Metamask".

**Comment by swayamarora5745** [comment, marked as answer]
I was supposed to deploy my code on the "Metamask" environment but I was deploying it on "Remix.
